serveradmin/common/static/js/serveradmin.js

```diff
@@ -11,38 +11,28 @@ $(document).ready(function() {
     window.spinner = {
         _running: 0,
         _spinner: $('#spinner'),
-        _toggle_css: function() {
-            if (this._spinner.css('display') === 'none') {
-                this._spinner.show();
-            }
-
-            if (this._spinner.hasClass('text-success')) {
-                this._spinner.removeClass('text-success');
-                this._spinner.addClass('text-secondary');
-
-                this._spinner.css('animation-play-state', 'paused');
-            }
-            else {
-                this._spinner.removeClass('text-secondary');
-                this._spinner.addClass('text-success');
-
-                this._spinner.css('animation-play-state', 'running');
-            }
-        },
         enable: function () {
-            this._running++;
-            if (this._running > 1) {
-                return;
-            }
+            console.debug('spinner enabled');
+            this._running = Date.now();
 
-            this._toggle_css();
+            this._spinner.removeClass('text-secondary');
+            this._spinner.addClass('text-success');
+            this._spinner.css('animation-play-state', 'running');
         },
         disable: function () {
-            this._running--;
+            console.debug('spinner disabled');
 
-            if (this._running === 0) {
-                this._toggle_css();
+            let elapsed = Date.now() - this._running;
+            if (elapsed > 1000) {
+                $('#spinner-timer-value').html(`${elapsed / 1000} s`);
```

Comment:
Just the seconds or seconds and milliseconds? Would it be ok from UX perspective if it would show `1` for `1.999` seconds?